### ISSUE TITLE
Missing dependency "Moment" on new and upgraded projects, while using OwlNativeDateTimeModule

### ISSUE THREAD

**Issue by florisvanderhaar** [danielmoncada/date-time-picker#125]
Even whilst not using Moment.js, there still appears to be a dependency for this date-time-picker. When you try to `npm start` a new or upgraded project, you will be hit with:
```
Error: The target entry-point "@danielmoncada/angular-datetime-picker" has missing dependencies:
 - moment
 ```

I was able to reproduce the error by creating a new angular app (with Angular Material) and setting up the imports:
```
@NgModule({
     imports: [ 
         OwlDateTimeModule, 
         OwlNativeDateTimeModule,
     ],
```

In projects where we used this date-time-picker, we have always used `OwlNativeDateTimeModule` and never the `OwlMomentDateTimeModule`, [as stated in the documentation](https://github.com/danielmoncada/date-time-picker#choose-a-date-implementation). I got the error after upgrading to Cypress 7, so it appears Cypress 6 was our last package that installed moment as a dependency.

As you might know already know, Moment.js is a legacy project. Reason enough to move away from it in new and upgraded projects, as apparently more projects are doing.

**Comment by GFlores98** [comment]
I solved installing moment.

`npm i moment`

**Comment by florisvanderhaar** [comment]
Yes, installing Moment.js is a valid workaround for the immediate short term. 

However, in doing so I will add a package to my application that's **not in active development** anymore, while I purposely configured the date-time-picker to use the `OwlNativeDateTimeModule` instead of Moment. This is a bad practice.

IMHO: The aim should always be to install the least amount of dependencies on a new project and avoid leaving any unwanted, unused or deprecated packages in a project during an upgrade.

Moment.js hadn't been deprecated yet when the date-time-picker was made. But still the documentation wrongly states:
> There are two pre-made modules:
> 
> OwlNativeDateTimeModule -- support for native JavaScript Date object
> OwlMomentDateTimeModule -- support for MomentJS object. **_(need to npm install ng-pick-datetime-moment and moment packages)_**

Source: https://danielykpan.github.io/date-time-picker/#locale-formats

TLDR; I shouldn't have to install a package I'm not using.

**Comment by EricDitp** [comment]
Same here, uselessly importing moment, which I don't really want.
I would like to remove moment from my project a.s.a.p.

thanks for all the good work.

**Comment by decline** [comment]
Having the same problem here. Even though I'm using `OwlNativeDateTimeModule` (and **not** `OwlMomentDateTimeModule`), I'm still forced to install Moment.js which will then also end up in my production-bundle, causing an increase of ~350kb. That's just bad 😕

**Comment by jcompagner** [comment]
i personally find it very weird that it still ends up in the production bundle
should angular treeshaking and so on take care of that?
Why would angular think it still references somehow?

**Comment by tmburnell** [comment]
I bet this is the reason we get this error:
https://github.com/danielmoncada/date-time-picker/blob/714f752d36184c158481f878a4327b16641e090e/projects/picker/src/public_api.ts#L19

I will do some more testing.

**Comment by tmburnell** [comment]
I did confirm if you drop 
```
export {
  MomentDateTimeAdapter,
  OwlMomentDateTimeAdapterOptions,
  OWL_MOMENT_DATE_TIME_ADAPTER_OPTIONS
} from './lib/date-time/adapter/moment-adapter/moment-date-time-adapter.class';
```
and
```
export { OwlMomentDateTimeModule } from './lib/date-time/adapter/moment-adapter/moment-date-time.module';
```
from the `public_api` it no longer errors saying moment is required.

**Comment by seowzhenjun0126** [comment]
Hi, is there any update on this? Thanks!

**Comment by danielmoncada** [comment]
@seowzhenjun0126  no updates, I can look more into it this week. 

In order to make the changes @tmburnell suggested, the "Moment Adapter Options" would need to be removed completely from this project, and ported over to a sub-project for those that are using them.

**Comment by tenonedesign** [comment]
Thanks for all your good work with this, @danielmoncada .  Freeing up 350k from the production bundle will be a big boost.

**Comment by danielmoncada** [comment]
Okay.. finally had time to remove the MomentAdapter, and extract it out into a separate project.

This is in the latest version (https://github.com/danielmoncada/date-time-picker/pull/143) v13.1.0.

If anyone does need a MomentJs adapter, you can find it here [npm](https://www.npmjs.com/package/@danielmoncada/angular-datetime-picker-moment-adapter) or [GitHub](https://github.com/danielmoncada/date-time-picker-moment-adapter).

**Comment by seowzhenjun0126** [comment]
Thank you so much for your work! Appreciate it

**Comment by danielmoncada** [comment]
@seowzhenjun0126  thanks!
